charts/delphai-deployment/templates/mapping.yaml

```
@@ -38,14 +42,67 @@ spec:
   service: "{{ $.Release.Name }}:{{ $.Values.ports.httpgateway.port }}"
   {{ else }}
   {{- fail "No HTTP ports found to publish" -}}
-  {{ end }}
+  {{ end }} {{/* if $.Values.ingress.oAuth */}}
 
   allow_upgrade:
     - websocket
 
   timeout_ms: 60000
   idle_timeout_ms: 500000
 
+{{ if $.Values.ingress.publicApiPrefix }}
+
+{{ if or
+  (not (hasPrefix "/" $.Values.ingress.publicApiPrefix))
+  (not (hasSuffix "/" $.Values.ingress.publicApiPrefix))
+}}
+{{- fail "The public API prefix must start and end with a slash" }}
+{{ end }}
+
+{{ $publicApiPrefix := print
+  ( ternary
+    ""
+    ( print "/" $.Release.Name )
+    ( index $.Values.buildMetadata.labels "com.delphai.image.release" )
+  )
+  ( trimSuffix "/" $.Values.ingress.publicApiPrefix )
+}}
```

Golang's idea is to code simple and stupid. Such complex expressions are rather unreadable.

Is that maybe clearer? (not sure)
```suggestion
{{ $branchPrefix := "" }}
{{ if not (index $.Values.buildMetadata.labels "com.delphai.image.release") }}
  {{ $branchPrefix = print "/" $.Release.Name }}
{{ end }} {{/* if not (index $.Values.buildMetadata.labels "com.delphai.image.release") */}}

{{ $publicApiPrefix := ( trimAll "/" $.Values.ingress.publicApiPrefix ) }}
```